SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -734,7 +734,7 @@ int64 USpatialActorChannel::ReplicateActor()
 
 				if (ObjectRef.IsValid())
 				{
-					OnSubobjectDeleted(ObjectRef, RepComp.Key());
+					OnSubobjectDeleted(ObjectRef, RepComp.Key(), RepComp.Value()->GetWeakObjectPtr());
```

Comment:
Hi Nic, this is the bit i'm a little confused with.  On this codepath, before we call OnSubobjectDeleted(), it is already established that the weak pointer is invalid.  Doesn't that mean that internally, OnSubobjectDeleted() will not find the correct entry in the ObjectReferenceMap?

Reply:
I understand this change also prevents a crash b/c we aren't implicitly constructing a TWeakObjectPtr from a dangling UObject pointer, just trying to verify if the abovementioned behaviour too.

Reply:
Also if you look at the overload== operator for FWeakObjectPtr, you can see it will match against any other invalid weak pointers:

	FORCEINLINE bool operator==(const FWeakObjectPtr& Other) const
	{
		return 
			(ObjectIndex == Other.ObjectIndex && ObjectSerialNumber == Other.ObjectSerialNumber) ||
			(!IsValid() && !Other.IsValid());
	}

Reply:
I see. So what I observed is merely chance, although to be compared in the first place, both objects would have to share the same hash anyway. So very low chance of confusion, but chance anyway.
I assumed, given the existence of TWeakObjectPtrMapKeyFuncs, that weak pointer maps would use this by default with the proper template specialization (it does not use the operator ==).
After checking, it uses the default key functions for weak pointers, which is really a bug from Epic IMO, as I disagree with Epic's recommendation to avoid WeakPointers as map keys.
I will change the map definition to use the right specialization for the key functions, and check if there are other instances of this in the GDK.

Reply:
Sounds good Nick, thanks